libs/features/mas/src/merch-card-collection.js

```diff
@@ -122,6 +132,24 @@ export class MerchCardCollection extends LitElement {
             ${this.footer}`;
     }
 
+    checkReady() {
+        let ticks;
+        return new Promise((resolve) => {
+            const checkReadyInterval = setInterval(() => {
+                const aemFragment = this.querySelector('aem-fragment');
```

Comment:
when a merch-card-collection from MAS Studio is consumed, is there a case `aem-fragment` would appear later?
suggestion:
mark `checkReady` async:
check `aem-fragment` does not exist,  return early
Promise.race  `aem-fragment.updateComplete` with a timeout Promise.
witch `setInterval` configuration below, we might lose up to 99ms. during the initialisation.